### PR TITLE
chore: add preload module for Webviews

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
UI will need a specific preload module that will provide bindings (API) for webviews

- [x] depends on https://github.com/containers/podman-desktop/pull/5592

### What issues does this PR fix or reference?

part of https://github.com/containers/podman-desktop/issues/5140

### How to test this PR?

unit tests provided

### CODE REVIEW

File: packages/preload-webview/src/webview-preload.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }, 0);
+  }
+
+  // build the function that will be exposed to the webview for getState/postMessage/setState
+  protected buildApi(): unknown {
+    return () => {
+      // initialize the state from the webview
+      let state: unknown = this.#webviewInfo?.state ?? {};
+      if (this.#acquiredApi) {
+        throw new Error('An instance of the VS Code API has already been acquired');
```
Comment (lstocchi) [inline packages/preload-webview/src/webview-preload.ts]: Should VS Code be replaced?

Comment (benoitf) [reply packages/preload-webview/src/webview-preload.ts]: fixed

**Comment by benoitf** [comment]
rebased

**Review by lstocchi — APPROVED** [review]

File: packages/preload-webview/src/webview-preload.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      });
+    };
+  }
+
+  protected getWebviews(): Promise<WebviewInfo[]> {
+    return this.ipcInvoke('webviewRegistry:listWebviews') as Promise<WebviewInfo[]>;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  protected ipcRendererOn(channel: string, listener: (event: IpcRendererEvent, ...args: any[]) => void) {
```
Comment (axel7083) [inline packages/preload-webview/src/webview-preload.ts:119]: Can't we use `unknown` here ?

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/preload-webview/src/webview-preload.ts (hunk 2 above)
Comment (benoitf) [reply packages/preload-webview/src/webview-preload.ts:119]: if it works for you I'll come back with a follow-up PR I think I had an issue initially with the spread arguments

Comment (benoitf) [reply packages/preload-webview/src/webview-preload.ts:119]: fixing there: https://github.com/containers/podman-desktop/pull/5603

**Pull request merged**